tests/exporters/test_ndvi.py

```diff
@@ -0,0 +1,104 @@
+# from pathlib import Path
+from unittest.mock import patch, MagicMock
+import urllib.request
+# import pytest
+import numpy as np
+
+from src.exporters import NDVIExporter
+
+
+class TestNDVIExporter:
+    def test_init(self, tmp_path):
+        e = NDVIExporter(tmp_path)
+
+        assert e.ndvi_folder.name == 'ndvi'
+        assert (tmp_path / 'raw' / 'ndvi').exists()
+
+    @patch('os.system', autospec=True)
+    def test_checkpointing(self, mock_system, tmp_path, capsys):
+        # checks we don't redownload files
+        exporter = NDVIExporter(tmp_path)
+
+        # setup the already downloaded file
+        test_filename = '1981/testy_test.nc'
+        (tmp_path / 'raw/ndvi/1981').mkdir(parents=True, exist_ok=True)
+        (tmp_path / f'raw/ndvi/{test_filename}').touch()
+
+        exporter.wget_file(test_filename)
+        captured = capsys.readouterr()
+
+        expected_stdout = f'testy_test.nc for 1981 already donwloaded!\n'
+        assert captured.out == expected_stdout, \
+            f'Expected stdout to be {expected_stdout}, got {captured.out}'
+        mock_system.assert_not_called(), 'os.system was called! Should have been skipped'
+
+    @patch('urllib.request.Request', autospec=True)
+    def test_get_filenames(self, request_patch, monkeypatch, tmp_path):
```

Comment:
This is the function that i am having problems with.

I have two calls using request_patch [here L32: `beautiful_soup_url()`](https://github.com/esowc/ml_drought/pull/73/files?file-filters%5B%5D=.py&hide-deleted-files=true#diff-7b98cb09687849a48b46dc5c38658ef5R32)

What I want is to mock the parsing of the second expected response:

```python
        # EXPECTED response for second page (1981 all .nc files)
```
to produce 2 expected URLs. But there's nothing specific about the 2 calls to the `beautiful_soup_url` function . it's almost as if i want to jump into the function `get_ndvi_url_paths ` [Here at L:68](https://github.com/esowc/ml_drought/pull/73/files?file-filters%5B%5D=.py&hide-deleted-files=true#diff-7b98cb09687849a48b46dc5c38658ef5R68)

Reply:
I'm not sure I understand - what are the 2 URLs you want to produce?

As it is, you pass the 2nd `expected_response` to beautiful soup. Then, when you do
```python
for yrs in soup.find_all('a')
```
`soup.find_all` returns None

Reply:
It looks like the mocking is behaving as expected though?

Reply:
I want to mock the parsing of the html into the URLpaths for each of the files 

the html / output of the `BeautfulSoup()` function is in the text below

```
        expected_response = '<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">\n<html>\n' \
            '<head>\n  <title>Index of' \
            '/data/avhrr-land-normalized-difference-vegetation-index/access/' \
            '1981</title>\n</head>\n <body>\n<h1>Index of' \
            '/data/avhrr-land-normalized-difference-vegetation-index/access/1981' \
            '</h1>\n<table><tr><th>&nbsp;</th><th><a' \
            'href="?C=N;O=D">Name</a></th><th><a href="?C=M;O=A">Last' \
            'modified</a></th><th><a href="?C=S;O=A">Size</a></th><th><a' \
            'href="?C=D;O=A">Description</a></th></tr><tr><th' \
            'colspan="5"><hr></th></tr>\n<tr><td valign="top">&nbsp;</td><td><a' \
            'href="/data/avhrr-land-normalized-difference-vegetation-index/access/">Parent' \
            'Directory</a></td><td>&nbsp;</td><td align="right">  -' \
            '</td><td>&nbsp;</td></tr>\n<tr><td valign="top">&nbsp;</td><td><a' \
            'href="AVHRR-Land_v005_AVH13C1_NOAA-07_19810624_c20170610041337.nc">' \
            'AVHRR-Land_v005_AVH13C1_NOAA-07_19810624_c20170610041337.nc</a></td><td' \
            'align="right">12-Jul-2019 10:37  </td><td align="right">' \
            '51M</td><td>&nbsp;</td></tr>\n<tr><td valign="top">&nbsp;</td><td><a' \
            'href="AVHRR-Land_v005_AVH13C1_NOAA-07_19810625_c20170610042839.nc">' \
            'AVHRR-Land_v005_AVH13C1_NOAA-07_19810625_c20170610042839.nc</a></td><td' \
            'align="right">12-Jul-2019 10:37  </td><td align="right">' \
            '59M</td><td>&nbsp;</td></tr>\n<tr><td valign="top">&nbsp;</td><td><a'

```